bundles/org.openhab.binding.linktap/src/main/resources/OH-INF/thing/channel-types.xml

```diff
@@ -0,0 +1,223 @@
+		<description>Active watering status</description>
+		<category>Water</category>
+		<state readOnly="false"/>
+	</channel-type>
+
+	<channel-type id="rf-linked-type">
+		<item-type>Switch</item-type>
+		<label>RF Linked</label>
+		<description>Is the device RF linked</description>
+		<category>Switch</category>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="flm-linked-type">
+		<item-type>Switch</item-type>
+		<label>FLM Linked</label>
+		<description>The device has a included flow meter</description>
+		<category>Switch</category>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="is-leak-type">
+		<item-type>Switch</item-type>
+		<label>High Flow Detected</label>
+		<description>Unusually high flow rate detected alert</description>
+		<category>Alarm</category>
+		<state readOnly="false"/>
+	</channel-type>
+
+	<channel-type id="is-clog-type">
+		<item-type>Switch</item-type>
+		<label>Low Flow Detected</label>
+		<description>Unusually low flow rate detected alert</description>
+		<category>Alarm</category>
+		<state readOnly="false"/>
+	</channel-type>
+
+	<channel-type id="fall-status-type">
+		<item-type>Switch</item-type>
+		<label>Fallen Status</label>
+		<description>The device has fallen</description>
+		<category>Alarm</category>
+		<state readOnly="false"/>
+	</channel-type>
+
+	<channel-type id="fail-status-type">
+		<item-type>Switch</item-type>
+		<label>Shutdown Value Failed</label>
+		<description>The device has failed to close the valve</description>
+		<category>Alarm</category>
+		<state readOnly="false"/>
+	</channel-type>
+
+	<channel-type id="final-segment-type">
+		<item-type>Switch</item-type>
+		<label>Final ECO Segment</label>
+		<description>In ECO mode this is true when the final ON watering on segment is running</description>
+		<category>Switch</category>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="signal-level-type">
+		<item-type unitHint="%">Number:Dimensionless</item-type>
+		<label>Signal Level</label>
+		<description>Reception Signal Strength</description>
+		<category>QualityOfService</category>
+		<state readOnly="true" pattern="%.0f %%"/>
+	</channel-type>
+
+	<channel-type id="battery-level-type">
+		<item-type unitHint="%">Number:Dimensionless</item-type>
+		<label>Battery Level</label>
+		<description>Battery Remaining Level</description>
+		<category>BatteryLevel</category>
+		<state readOnly="true" pattern="%.0f %%"/>
+	</channel-type>
+
+	<channel-type id="water-cut-type">
+		<item-type>Switch</item-type>
+		<label>Water Cutoff</label>
+		<description>Water cut-off alert</description>
+		<category>Alarm</category>
+		<state readOnly="false"/>
+	</channel-type>
+
+	<channel-type id="flow-rate-type">
+		<item-type unitHint="l/min">Number:VolumetricFlowRate</item-type>
+		<label>Flow Rate</label>
+		<description>Current water flow rate</description>
+		<category>Flow</category>
+		<state readOnly="true" pattern="%.2f %unit%"/>
```

Comment:
Hi, yes the protocol appears to always encode to 2 decimal places the values so that's why its been setup to that level. Whether the devices have a higher precision and round to keep the payload's smaller I am not sure (likely as there seems to be a packet limit in place), but its the protocol limit that effectively has set the precision to 2 dp. This is the same for all channels set to 2 dp.

Reply:
To what unit is this 2 decimal value, m3 or liters?

Reply:
The API defaults when its shipped to using liter's, however users can switch the configuration to work in m3. So most will likely use litre's but m3 is possible, it they find that option.